internal/provider/content_translation_resource.go

```diff
@@ -0,0 +1,214 @@
+		ctx,
+		"multipart/form-data; boundary="+writer.Boundary(),
+		strings.NewReader(body.String()),
+	)
+
+	return checkMetabaseResponse(uploadResp, err, []int{200}, "upload content translation dictionary")
+}
+
+func (r *ContentTranslationResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "A Metabase content translation dictionary. This resource manages the translation dictionary for Metabase Enterprise Edition. The dictionary content is stored as a hash in the state for efficiency.\n\n## ⚠️ Important Warning\n\n**Deleting this resource will erase all translation dictionaries on the Metabase instance.** When you destroy this resource, it uploads an empty dictionary to Metabase, effectively removing all translations. Make sure to backup your translation data before destroying this resource if you need to preserve the translations.",
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				MarkdownDescription: "A unique identifier for the translation set.",
+				Computed:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"dictionary": schema.StringAttribute{
+				MarkdownDescription: "The CSV content of the translation dictionary. Must have columns: Locale Code (locale code), String (text to translate), Translation (translated text). Example: `Locale Code,String,Translation\\npt-BR,Examples,Exemplos\\nen,Dashboard,Dashboard`",
+				Required:            true,
+			},
+			"content_hash": schema.StringAttribute{
+				MarkdownDescription: "SHA256 hash of the dictionary content, used for change detection and state management.",
+				Computed:            true,
+			},
+		},
+	}
+}
+
+func (r *ContentTranslationResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data *ContentTranslationResourceModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Upload the translation dictionary
+	resp.Diagnostics.Append(r.uploadContentTranslationDictionary(ctx, data.Dictionary.ValueString())...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Update the model with computed values
+	resp.Diagnostics.Append(updateModelFromContentTranslation(data.Dictionary.ValueString(), data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Save data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *ContentTranslationResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data *ContentTranslationResourceModel
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// If we have a dictionary in state, verify it's still current by checking the hash
```

Comment:
Well, if I remember correctly, Terraform simply compares the existing state with what is being set in this function. So while it will check the hash... it will also check the full dictionary.
Am I missing something?